db/migrate/20150924082915_create_user_project_roles.rb

```diff
@@ -0,0 +1,8 @@
+class CreateUserProjectRoles < ActiveRecord::Migration
```

Comment:
If you go with the has_and_belongs_to_many assoc, you'll have to rename this join table to project_role_users I think, as Rails alphabetises the tables